src/demo_parser.rs

```diff
@@ -122,6 +133,53 @@ pub fn parse_directory<'a>(
     )(i)
 }
 
+/// Parse a fallback directory for demo files that were not finalized by a client.
+pub fn parse_fallback_directory<'a>(
+    frames_start: &'a [u8],
+    file_start: &'a [u8],
+    should_parse_netmessage: bool,
+    aux: AuxRefCell,
+) -> Result<'a, Directory> {
+    let file_length = frames_start.len() as i32;
+    let frame_offset = (file_start.len() - frames_start.len()) as i32;
+
+    let mut frames: Vec<Frame> = vec![];
+    let mut frames_start = &frames_start[..];
+
+    loop {
+        match parse_frame(frames_start, should_parse_netmessage, aux.clone()) {
+            Ok((end_current_frame, frame)) => {
+                frames_start = end_current_frame;
+                frames.push(frame);
+
+                if end_current_frame.is_empty() {
+                    break;
+                }
+            }
+            Err(_) => break,
+        }
+    }
```

Comment:
You can make this idiomatic nom by using [many0](https://docs.rs/nom/latest/nom/multi/fn.many0.html) nom combinator. `many0` will parse until there is error and collect the result in a vector.

```suggestion
    let parser = |i| parse_frame(i, should_parse_netmessage, aux.clone());
    let (end_of_frames, frames) = many0(parser)(frames_start)?;
```